buildman/component/buildcomponent.py

```diff
@@ -360,19 +362,26 @@ def _on_log_message(self, phase, json_data):
                 # If the json data contains an error, then something went wrong with a push or pull.
                 if "error" in log_data:
                     yield From(self._build_status.set_error(log_data["error"]))
+        except RedisError as re:
+            # Move forward with the build if it is a Redis error
+            logger.warning("Failed to set phase on log for build ID: %s - %s", build_id, bse)
+            build_model.update_phase_then_close(build_id, phase)
```

Comment:
@jzelinskie Sure, but it was already yielding control of the event loop to that same synchronous call whenever the worker logged a phase change in `buildstatus.py`.

Since that database call is coupled with the Redis call in `buildstatus.py`, if Redis is down, the build phase won't get updated correctly. 

In this case, we're making sure that we allow the build to move forward, or marked as failure if Redis is not available.

Reply:
Ideally, we'd wrap this with the same async wrapper we use for etcd, but I don't think it's worth the effort (and complexity) if we can get by with this until Python 3 or a new build system.